src/APIClient/index.tsx

```diff
@@ -20,6 +21,10 @@ const DEFAULT_API_CLIENT_CONFIG: APIClientConfiguration = {
     },
 };
 
+const generateUploadTaskId = () =>
+    Math.random().toString(36).slice(-10) +
+    Math.random().toString(36).slice(-10);
+
```

Comment:
From what I understand RN is essentially single-threaded so there will only ever be a single upload at a time when it's called, I don't think this will be needed?

Reply:
@shazm JS may be single threaded but the native side is not, right?